src/coreclr/jit/ee_il_dll.cpp

```diff
@@ -1122,9 +1122,39 @@ void Compiler::eeDispLineInfos()
  * (e.g., host AMD64, target ARM64), then VM will get confused anyway.
  */
 
+void Compiler::eeAllocMem(AllocMemArgs* args, UNATIVE_OFFSET hotSizeRequest, UNATIVE_OFFSET coldSizeRequest)
+{
+#ifdef DEBUG
+    // Fake splitting implementation: hot section = hot code + 4K buffer + cold code
+    const UNATIVE_OFFSET buffer = 4096;
+    if (JitConfig.JitFakeProcedureSplitting() && (coldSizeRequest > 0))
+    {
+        args->hotCodeSize  = hotSizeRequest + buffer + coldSizeRequest;
+        args->coldCodeSize = 0;
```

Comment:
Note that changing `args` fields works out because the caller doesn't (I think) consult these values afterwards. To be perfectly clean, you might want to copy `args` to a local temp before modifying its fields.